### PR TITLE
Fix bug where Spanish words do not have spacing in between. 

### PR DESCRIPTION
**Pull request by hollyyuqizheng**

### CODE REVIEW

File: data/index.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
       "con-Latn-EC": "103 Singo a'i",
       "en": "Black People",
       "es": "El negro",
-      "_default": "Singo a'i"
+      "_default": "103"
```
Comment (sciepsilon) [inline data/index.json:9]: This change is incorrect - "Singo a'i" is a much better title for LingView to display than "103". On the old files, we used the edit.js script to modify some of the metadata that LingView displays, including this title, and it looks like those changes got lost as you removed and re-uploaded the files. It should be safe (and correct) to undo all the changes to this index.json file.

File: data/json_files/1ed3d641-acd9-4466-811d-17c8ed59844c.json [hunk 2]
```diff
@@ -3,19 +3,19 @@
     "timed": true,
     "story ID": "1ed3d641-acd9-4466-811d-17c8ed59844c",
     "title": {
-      "_default": "Kuke chiste"
+      "_default": "20170804_kuke_chiste_FACQ"
```
Comment (sciepsilon) [inline data/json_files/1ed3d641-acd9-4466-811d-17c8ed59844c.json:6]: Same thing here - please revert all changes to this file

File: data/json_files/1e3bff88-29b9-4b3f-be34-bffc38624bac.json [hunk 3]
```diff
@@ -821,16 +841,16 @@
...
             }
           ]
         }
       ]
     },
     {
-      "num_slots": 38,
-      "text": "Tsonsi tsama attepa yaya que cundyi'fa tsu sin'an qquen susi tise yaya in'jan, cuintsu tsa'caen injengembe camba tisu yaya cundyi'je'fama atte'ta singo a'ive daye.",
+      "num_slots": 40,
+      "text": "Tsonsi tsama attepa yaya que cundyi'fa tsu sin'an qquen susi tise yaya in'jan , cuintsu tsa'caen injengembe camba tisu yaya cundyi'je'fama atte'ta singo a'ive daye .",
```
Comment (sciepsilon) [inline data/json_files/1e3bff88-29b9-4b3f-be34-bffc38624bac.json:853]: Uh-oh - there shouldn't be spaces added before these commas!

File: data/json_files/1e3bff88-29b9-4b3f-be34-bffc38624bac.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
       "con-Latn-EC": "103 Singo a'i",
       "en": "Black People",
       "es": "El negro",
-      "_default": "Singo a'i"
+      "_default": "103"
```
Comment (sciepsilon) [inline data/json_files/1e3bff88-29b9-4b3f-be34-bffc38624bac.json:9]: Same thing here - please undo the change from "Singo a'i" to "103" and the change in date_uploaded

File: preprocessing/preprocess_flex.js [hunk 5]
```diff
@@ -26,8 +26,17 @@ function isSeparator(char) {
...
+// to detect in this file, so this word should not be marked as a punctuation. 
 function isPunctuation(word) {
-  return word.item[0].$.type === "punct";
+  if (word.item[0].$.type === "punct")  {
+    const wordElement = word.item[0]._;
+    if (wordElement && wordElement.match(/^[0-9a-zA-ZÀ-ÿ]+$/)) {
+      return false; 
+    } 
+  }
+  return false;
```
Comment (sciepsilon) [inline preprocessing/preprocess_flex.js:39]: This line should be "return true", right?

File: preprocessing/preprocess_flex.js [hunk 6]
```diff
@@ -26,8 +26,17 @@ function isSeparator(char) {
...
 //   structured as in FLEx
 // return true if the word is tagged as punctuation within FLEx
+// If the word is marked as type "punct" but the word is alphanumeric,
+// this word might come from a different language than what FLEx is trying 
+// to detect in this file, so this word should not be marked as a punctuation. 
 function isPunctuation(word) {
-  return word.item[0].$.type === "punct";
+  if (word.item[0].$.type === "punct")  {
+    const wordElement = word.item[0]._;
+    if (wordElement && wordElement.match(/^[0-9a-zA-ZÀ-ÿ]+$/)) {
```
Comment (sciepsilon) [inline preprocessing/preprocess_flex.js:35]: Regexes are always hard to read, so it might be worth an additional comment here

File: preprocessing/preprocess_flex.js (hunk 5 above)
Comment (hollyyuqizheng) [reply preprocessing/preprocess_flex.js:39]: Ah I see, there needs to be an else statement after the if condition.

File: data/index.json (hunk 1 above)
Comment (hollyyuqizheng) [reply data/index.json:9]: Updated.

File: data/json_files/1ed3d641-acd9-4466-811d-17c8ed59844c.json (hunk 2 above)
Comment (hollyyuqizheng) [reply data/json_files/1ed3d641-acd9-4466-811d-17c8ed59844c.json:6]: Updated.

File: preprocessing/preprocess_flex.js (hunk 6 above)
Comment (hollyyuqizheng) [reply preprocessing/preprocess_flex.js:35]: Updated.

File: .gitignore [hunk 7]
```diff
@@ -1,3 +1,5 @@
+data/.DS_Store
+.DS_Store
```
Comment (sciepsilon) [inline .gitignore:2]: Let's replace these two lines with: 
**/.DS_Store
to exclude .DS_Store files from all directories

File: preprocessing/preprocess_flex.js (hunk 6 above)
Comment (sciepsilon) [reply preprocessing/preprocess_flex.js:38]: Thanks!

File: preprocessing/preprocess_flex.js (hunk 5 above)
Comment (sciepsilon) [reply preprocessing/preprocess_flex.js:44]: Looks good!

File: .gitignore (hunk 7 above)
Comment (hollyyuqizheng) [reply .gitignore]: Updated.
